### DISCUSSION TITLE
Printing negative exponents in laurent polynomials

### DISCUSSION THREAD

**Discussion by JohnAAbbott** [oscar-system/Oscar.jl#2384] [General]
Currently negative exponents are printed without brackets: _e.g.  `2*x^-3`.
I would prefer negative exponent to be printed in brackets: _e.g._  `2*x^(-3)`

My reasons: it is easier to read, and less risk of a small typo being misinterpreted: _e.g_  `2*x^2-3`

**Comment by thofma** [comment]
I disagree that this should be changed. 

```
julia> f= x^10 + x^-1 + x^-2 + x^-3
x^10 + x^-1 + x^-2 + x^-3
```

Is this hard to read? In my opinion parentheses add too much useless noise.

Can you elaborate about the small typo being misinterpreted?

**Reply by JohnAAbbott** [reply]
There is a 1 character difference between `2*x^2-3` and `2*x^-3`;  it could easily be that I meant to type the first, but accidentally typed the second.  I have myself made a similar typo (a few years ago); it anyway triggered an error because a negative exponent was not allowed in the context I was using then).

**Reply by thofma** [reply]
You can always enter them with parentheses. I thought the proposal was about printing?

**Reply by JohnAAbbott** [reply]
My proposal would be to make the printed form as shown here:
```
julia> f= x^10 + x^-1 + x^-2 + x^-3
x^10 + x^(-1) + x^(-2) + x^(-3)
```
I find this printed form nicer to read.  I do agree that it is largely a matter of personal preference.  There might be some relevance if one wanted to copy-and-paste the printed form from Oscar into another system (_e.g._ CoCoA) which does require brackets around negative exponents -- of course, using `sed` with a suitable regexp will effect the necessary transformation.

**Reply by thofma** [reply]
Yes, I understood that part. What I don't understand is the connection to the small typo that you mentioned.

P.S.: There are a dozens of languages where we could try to support copy-and-paste. It is not worth the hassle to start with this.

**Reply by JohnAAbbott** [reply]
The syntax for reading is no doubt imposed upon us by Julia, so we cannot change that.  However we could encourage users to write brackets around negative exponents also for input: then it is less likely that one would mistype `2*x^2-3` as `2*x^(-3)` as there are three(=1 deletion + 2 insertions) character differences, rather than the 1 character difference highlighted above.

**Reply by JohnAAbbott** [reply]
And by _printing_ with brackets we might hope to **encourage** users to use them also _for input._

**Comment by JohnAAbbott** [comment]
Strictly a different issue: shouldn't the constructor be called `laurent_polynomial_ring` rather than `LaurentPolynomialRing`?
